internal/failure_domain.go

```diff
@@ -87,6 +87,23 @@ func PickFewest(failureDomains clusterv1.FailureDomains, machines FilterableMach
 	return pointer.StringPtr(aggregations[0].id)
 }
 
+// NPickFewest returns N failure domains with the fewest number of machines.
+func NPickFewest(failureDomains clusterv1.FailureDomains, machines FilterableMachineCollection, replicas int) []string {
+	aggregations := pick(failureDomains, machines)
+	if len(aggregations) == 0 {
+		return nil
+	}
+	sort.Sort(aggregations)
+	fds := make([]string, 0)
+	for i := 0; i < len(aggregations); i++ {
+		fds = append(fds, aggregations[i].id)
+	}
+	if len(aggregations) <= replicas {
+		return fds[:len(aggregations)]
+	}
+	return fds
```

Comment:
I believe this will end up being logically the same as just `return fds`, because the length of the aggregations and fds slices will always be the same length, so `fds[:len(aggregations)]` ends up being the same as the entire slice of fds.

Reply:
this will actually turn into
```
if len(aggregations) > replicas {
		return fds[:replicas]
	}
	return fds
```
since we want the least num of fds for the replicas

Reply:
what about just making it `return fds[:replicas]` and putting a guard above it (i.e. `if len(aggregations) == 0 ...`)?

Reply:
I believe that guard is already present. Regardless, wouldn't this cause slice bounds out-of-range. For example, suppose we want 5 replicas, but 3 failure domains are available (us-east-1a, us-east-1b, us-east-1c).

Reply:
There is no guard present, again it would check if `aggregations` is empty.  The purpose of the guard is to prevent slicing out-of-bounds.

Reply:
I apologize, I meant in this case a guard for replicas (the value being used in creating the slice).  I have an alternative though that would change how this works slightly presented in a different conversation thread above, check that out.